lib/samples/analyze_network_with_subnetwork_trace/analyze_network_with_subnetwork_trace.dart

```diff
@@ -0,0 +1,691 @@
+                          Expanded(
+                            flex: 2,
+                            child: ElevatedButton(
+                              onPressed: _configuration != null
+                                  ? _onReset
+                                  : null,
+                              child: const Text('Reset'),
+                            ),
+                          ),
+                          const SizedBox(width: 8),
+                          Expanded(
+                            flex: 3,
+                            child: ElevatedButton(
+                              onPressed: _ready && _canAddCondition()
+                                  ? _onAddCondition
+                                  : null,
+                              child: const Text('Add Condition'),
+                            ),
+                          ),
+                          const SizedBox(width: 8),
+                          Expanded(
+                            flex: 2,
+                            child: ElevatedButton(
+                              onPressed: _startingLocation != null
+                                  ? _onRunTrace
+                                  : null,
+                              child: const Text('Trace'),
+                            ),
+                          ),
+                        ],
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+            // Loading indicator.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+    );
+  }
+
+  // Performs important tasks including loading utility network and setting trace parameters.
+  Future<void> _setup() async {
+    try {
+      await _setupTraceParameters();
+    } finally {
+      setState(() => _ready = true);
+    }
+  }
+
+  // Loads the utility network and sets the trace parameters and other information.
+  Future<void> _setupTraceParameters() async {
+    // Constants for creating the default trace configuration.
+    const domainNetworkName = 'ElectricDistribution';
+    const tierName = 'Medium Voltage Radial';
+
+    // Constants for creating the default starting location.
+    const deviceTableName = 'Electric Distribution Device';
+    const assetGroupName = 'Circuit Breaker';
+    const assetTypeName = 'Three Phase';
+    const globalIdString = '1CAF7740-0BF4-4113-8DB2-654E18800028';
+
+    // Load the service geodatabase.
+    final serviceGeodatabase = ServiceGeodatabase.withUri(
+      Uri.parse(
+        'https://sampleserver7.arcgisonline.com/server/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer/0',
+      ),
+    );
+    await serviceGeodatabase.load();
+
+    // Load the utility network.
+    _utilityNetwork = UtilityNetwork(serviceGeodatabase);
+    await _utilityNetwork!.load();
+
+    final definition = _utilityNetwork!.definition;
+    if (definition == null) return;
```

Comment:
I think we should use `!` like this:
```dart
final definition = _utilityNetwork!.definition!;
```
We know that our utility network has a definition. If it doesn't for some reason, we would actually prefer a crash rather than to "return" with no error message that leaves the UI with a "loading" indicator forever.